src/main/java/by/unvisiblee/questionnaireApp/controller/AuthController.java

```diff
@@ -0,0 +1,41 @@
+package by.unvisiblee.questionnaireApp.controller;
+
+import by.unvisiblee.questionnaireApp.dto.AuthResponse;
+import by.unvisiblee.questionnaireApp.dto.LoginRequest;
+import by.unvisiblee.questionnaireApp.dto.RegisterRequest;
+import by.unvisiblee.questionnaireApp.service.AuthService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.Errors;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/api/auth")
+@Validated
+public class AuthController {
+
+    @Autowired
+    private AuthService authService;
+
+    @PostMapping("/signup")
+    public ResponseEntity<String> signup(@RequestBody @Valid final RegisterRequest registerRequest) {
+        authService.signup(registerRequest);
+        return new ResponseEntity<>("User Registration Successful",
+                HttpStatus.OK);
+    }
+
+    @GetMapping("/verification/{token}")
+    public ResponseEntity<String> verifyAccount(@PathVariable String token) {
+        authService.verifyAccountByToken(token);
+        return new ResponseEntity<>("User verified successfully!", HttpStatus.OK);
+    }
+
+    @PostMapping("/login")
+    public AuthResponse login(@Valid @RequestBody LoginRequest loginRequest) {
+        return authService.login(loginRequest);
```

Comment:
Честно говоря единственное замечание, которое не понял.
 Дело в том, что я возвращаю только хеш + юзернем и больше ничего, а должно быть как по ссылке?

Reply:
Stormpath recommends that you store your JWT in cookies for web applications, because of the additional security they provide, and the simplicity of protecting against CSRF with modern web frameworks. HTML5 Web Storage is vulnerable to XSS, has a larger attack surface area, and can impact all application users on a successful attack.

В целом, для учебного приложения это не критично. Но знать нужно - cookie более безопасны, хотя тоже не на 100%.

Reply:
То есть с бэкенда токен при успешном логине мы отправляем не в body, а в cookies?